projects/observability/src/shared/components/cartesian/cartesian-chart.component.ts

```diff
@@ -93,7 +98,10 @@ export class CartesianChartComponent<TData> implements OnChanges, OnDestroy {
         this.chartTooltipBuilderService.constructTooltip<TData, Series<TData>>(data =>
           this.convertToDefaultTooltipRenderData(data)
         )
-      );
+      )
+      .withEventListener(ChartEvent.Select, selectedData => {
+        this.selectionChange.emit(selectedData as CartesianSelectedData<TData>);
```

Comment:
Is this cast required? Can we type it so that `selectedData` come as         `CartesianSelectedData<TData>` ?

Reply:
may be we can do `if(data instanceOf CartesianSelectedData<TData>)` ?

Reply:
@jyothishjose6190 looks like you missed this